File: model/src/main/java/com/nigeleke/cribbage/suit/Face.java
package com.nigeleke.cribbage.suit;

public enum Face {
    Ace(1, 1),
    Two(2, 2),
    Three(3, 3),
    Four(4, 4),
    Five(5, 5),
    Six(6, 6),
    Seven(7, 7),
    Eight(8, 8),
    Nine(9, 9),
    Ten(10, 10),
    Jack(10, 11),
    Queen(10, 12),
    King(10, 13);

    public final int value;
    public final int rank;

    private Face(int value, int rank) {
        this.value = value;
        this.rank = rank;
    }
}

File: packages/domain/src/card/face.rs
use std::ptr::write;

use strum::EnumIter;

use super::{Rank, Value};

/// The face of a playing card (Ace through King).
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, EnumIter)]
#[rustfmt::skip]
pub enum Face {
    #[doc(hidden)] Ace,
    #[doc(hidden)] Two,
    #[doc(hidden)] Three,
    #[doc(hidden)] Four,
    #[doc(hidden)] Five,
    #[doc(hidden)] Six,
    #[doc(hidden)] Seven,
    #[doc(hidden)] Eight,
    #[doc(hidden)] Nine,
    #[doc(hidden)] Ten,
    #[doc(hidden)] Jack,
    #[doc(hidden)] Queen,
    #[doc(hidden)] King,
}

impl Face {
    /// Returns the rank used for ordering (Ace low, King high).
    #[inline]
    pub fn rank(&self) -> Rank {
        Rank::from(self)
    }

    /// Returns the point or face value.
    ///
    /// - Ace = 1 (low)
    /// - 2–10 = face value
    /// - Jack/Queen/King = 10
    #[inline]
    pub fn value(&self) -> Value {
        Value::from(self)
    }
}

impl std::fmt::Debug for Face {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_str(match self {
            Face::Ace => "A",
            Face::Two => "2",
            Face::Three => "3",
            Face::Four => "4",
            Face::Five => "5",
            Face::Six => "6",
            Face::Seven => "7",
            Face::Eight => "8",
            Face::Nine => "9",
            Face::Ten => "T",
            Face::Jack => "J",
            Face::Queen => "Q",
            Face::King => "K",
        })
    }
}
